test/fetch/integrity.js

```diff
@@ -14,49 +15,49 @@ setGlobalDispatcher(new Agent({
   keepAliveMaxTimeout: 1
 }))
 
-test('request with correct integrity checksum', (t) => {
+test('request with correct integrity checksum', (t, done) => {
   const body = 'Hello world!'
   const hash = createHash('sha256').update(body).digest('base64')
 
   const server = createServer((req, res) => {
     res.end(body)
   })
 
-  t.teardown(server.close.bind(server))
+  t.after(server.close.bind(server))
 
   server.listen(0, async () => {
     const response = await fetch(`http://localhost:${server.address().port}`, {
       integrity: `sha256-${hash}`
     })
-    t.strictSame(body, await response.text())
-    t.end()
+    assert.strictEqual(body, await response.text())
+    done()
   })
 })
 
-test('request with wrong integrity checksum', (t) => {
+test('request with wrong integrity checksum', (t, done) => {
   const body = 'Hello world!'
   const hash = 'c0535e4be2b79ffd93291305436bf889314e4a3faec05ecffcbb7df31ad9e51b'
 
   const server = createServer((req, res) => {
     res.end(body)
   })
 
-  t.teardown(server.close.bind(server))
+  t.after(server.close.bind(server))
 
   server.listen(0, () => {
     fetch(`http://localhost:${server.address().port}`, {
       integrity: `sha256-${hash}`
     }).then(response => {
-      t.pass('request did not fail')
+      assert.ok(true)
```

Comment:
this test isn't equal - it doesn't look like it will fail if fetch throws an error here?

Reply:
looking at this test, is this meant to be t.fail? We pass if fetch rejects and resolves...?

Reply:
@KhafraDev

You're right, we are expecting it to fail, so if it passes, we should fail. I think it was meant to be t.fail.
Should I change the `assert.ok(true)` to something like `throw new Error('request did not fail')`?

Reply:
sounds good

Reply:
@KhafraDev Updated the test to use `assert.rejects` and match the error, think this is better. Wdyt?